libs/vkd3d/command.c

```diff
@@ -1412,35 +1421,117 @@ static HRESULT STDMETHODCALLTYPE d3d12_shared_fence_GetDevice(d3d12_fence_iface
 static UINT64 STDMETHODCALLTYPE d3d12_shared_fence_GetCompletedValue(d3d12_fence_iface *iface)
 {
     struct d3d12_shared_fence *fence = shared_impl_from_ID3D12Fence1(iface);
-    const struct vkd3d_vk_device_procs *vk_procs = &fence->device->vk_procs;
+
+    TRACE("iface %p\n", iface);
+    return vkd3d_atomic_uint64_load_explicit(&fence->last_value, vkd3d_memory_order_acquire);
+}
+
+static void vkd3d_shared_fence_signal_events(struct d3d12_shared_fence *fence, uint64_t value)
```

Comment:
Is this even needed? If the shared fence thread waits for a semaphore value, it should be unblocked when the vkSignalSemaphore is called on host.